### ISSUE TITLE
Some chunks fail load when server render distance exceeds client's

### ISSUE THREAD

**Issue by Arctrometa** [GeyserMC/Geyser#3809]
### Describe the bug

After exploring the world for a bit of time the chunks in the player's render distance just stop rendering at a certain point, all blocks are still intractable but they don't show up in the mobile player's client. Only as a empty void.

### To Reproduce

1. Joined using a mobile client on 1.19.83
2. Flew around the world for approx. 1 minute or so.
3. New chunks stop visually loading.

### Expected behaviour

The chunks should show up for the player.

### Screenshots / Videos

![Screenshot_2023-06-02-13-46-43-451_com mojang minecraftpe](https://github.com/GeyserMC/Geyser/assets/91560943/d29f54be-6d16-49ca-bf6b-45fc6f7064c2)

### Server Version and Plugins

The server is on Geyser-Spigot with Floodgate, Geyser and Chunky.

### Geyser Dump

https://dump.geysermc.org/DtNeCCCv9d0sRLT0EYaQcUmOjPRedtQc

### Geyser Version

 2.1.0-SNAPSHOT (git-master-246ebdd)

### Minecraft: Bedrock Edition Device/Version

1.19.83 Mobile

### Additional Context

  Confirmed to not occur on 1.19.83 PC.

**Comment by Vortecsz** [comment]
Same problem, but with fabric

**Comment by Vortecsz** [comment]
After some tests, the problem seems to be Lithium or Phosphor in my case. Do you guys know if i can do something?

**Comment by chandlerpl** [comment]
Same problem here with Geyser Standalone version 2.1.0-SNAPSHOT (git-DEV-f7d2b4e)

**Comment by Kas-tle** [comment]
What do you guys have your server render distance set to?

**Comment by Kas-tle** [comment]
Ok so on further destination this is not actually specific to mobile clients, but rather specific to low render distances. If I set my render distance to 6 on desktop, which is the default for many mobile devices. I also experience this. Setting it to 10 on mobile (max) likewise seems to eliminate the issue. We may just have to do some kind of padding when lower render distances are used.

**Comment by Kas-tle** [comment]
It's likely a function of how close your render distance is to the server render distance.

**Comment by Kas-tle** [comment]
@Arctrometa for my own sanity could you try this without chunky? From reading the plugin description it doesn't seem like it should affect actual chunk loading, but I'm not really sure. I can't seem to reproduce this on the latest paper like you have, but I can on older versions with viaversion.

**Comment by Arctrometa** [comment]
> @Arctrometa for my own sanity could you try this without chunky? From reading the plugin description it doesn't seem like it should affect actual chunk loading, but I'm not really sure. I can't seem to reproduce this on the latest paper like you have, but I can on older versions with viaversion.

Sure i'll have a look.

**Comment by EpicJosch** [comment]
Same problem. Whats the current state of this issue?

**Comment by Arctrometa** [comment]
> @Arctrometa for my own sanity could you try this without chunky? From reading the plugin description it doesn't seem like it should affect actual chunk loading, but I'm not really sure. I can't seem to reproduce this on the latest paper like you have, but I can on older versions with viaversion.

I've tried running the server without chunky and the same problem occurred only on mobile devices.

**Comment by EpicJosch** [comment]
I was also running it without chunky and same problem occurred. Then tried with chunky and no changes to the problems behavior

**Comment by moemeow233** [comment]
same problem paper1.17.1, geyser+viaversion view distance 10, bedrock 1.20.1

**Comment by moemeow233** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by Bill-Joe-Tech** [comment]
Having the same issue on 1.20.1 (Spigot, GeyserMC + Floodgate). Seems to affect <ins>Bedrock editions</ins> (including PE/mobile), and <ins>**not** Java</ins>.

Temporary Fixes: 
- Reload world & wait for chunks to load.
- Increase render distance to 7+ (assuming unchanged server chunk settings) & reload world.
- Running GeyserMC in a proxy such as velocity or BungeeCord.

Sources:
1. According to @_jiimmy. ([Discord Thread](https://discord.com/channels/613163671870242838/1126202314634571868/1126202314634571868)), “Apparently this error does not occur if I run the geysermc plugin in some proxy like velocity or bungecord.”
2. From above, @Kas-tle ([Comment is above](https://github.com/GeyserMC/Geyser/issues/3809#issuecomment-1576021883)).

**Comment by YaBoiEddyC** [comment]
This issue depends on the device, I've tested a couple of devices on my lan server (with and without the same network). I tested both paper and fabric for this. PaperMC doesn't have this issue probably because of the chunk rewrite. FabricMC does though and like I said for me it seems to be on what device you're using, here's my list. (You can rejoin everytime chunks stop loading and it'll fix for a bit that's what works best for me. just rejoin everytime chunks stop loading and it should be just fine for a while.)

1. A windows laptop with Ryzen 5 5600h and 3060 laptop GPU 16gbs of tam. Doesn't happen on this device unless I'm connected to another network. (The bedrock version of Java)

2. A android phone with Snapdragon Gen 1 with 8gbs. occurs on both lan and other networks

3. Another android phone with Snapdragon 778g and 6gbs of ram. this one somehow doesn't have the issue and it only happens if I turn off wifi mid game and rejoin again.

4. A android Tablet with exynoss 7904 and 3gbs of ram. It happens on both same network and on another network

5. My friends IPad Pro 3rd gen, It somehow doesn't happen to him on both same network and on his wifi.

6. A windows device with a core i7 5500u and 32gbs of ram. The chunks do load on lan but really slowly (probably my device lol) and on another network it sometimes sends me into the void or chunks don't load at all.

7. An iPhone 12 the regular version, Chunks will only load on another network for some reason and stop after loading a few chunks on my own network that the server is connected to

8. 2 more android devices (one with a Snapdragon 665 and 4gbs of ram and a exynoss 850 with 4gbs of ram) the results for these 2 were the same as device Number 2 I tested.

9. Xbox series X and PS4. The chunks load correctly and smoothly on this and another network with the only issue being it has a bit of lag spikes (that might just be me on 2.4g wifi though)

10 Nintendo switch. Performance is exceptional (probably just the switch not being well optimized) chunks wont load when connected on another network or it drops me in an empty unloaded place however on Lan it will load just fine.

11. An old hp 15 notebook with a Core i3 5600u and Intel HD graphics 16gbs of ram. Chunks load fine on lan and surprisingly on wireless 2g it does better. When connected to another network it doesn't load some chunks and skips some chunks too.

**Comment by Kas-tle** [comment]
> This issue depends on the device, I've tested a couple of devices on my lan server (with and without the same network). I tested both paper and fabric for this. PaperMC doesn't have this issue probably because of the chunk rewrite. FabricMC does though and like I said for me it seems to be on what device you're using, here's my list. (You can rejoin everytime chunks stop loading and it'll fix for a bit that's what works best for me. just rejoin everytime chunks stop loading and it should be just fine for a while.)
> 
> 1. A windows laptop with Ryzen 5 5600h and 3060 laptop GPU 16gbs of tam. Doesn't happen on this device unless I'm connected to another network. (The bedrock version of Java)
> 2. A android phone with Snapdragon Gen 1 with 8gbs. occurs on both lan and other networks
> 3. Another android phone with Snapdragon 778g and 6gbs of ram. this one somehow doesn't have the issue and it only happens if I turn off wifi mid game and rejoin again.
> 4. A android Tablet with exynoss 7904 and 3gbs of ram. It happens on both same network and on another network
> 5. My friends IPad Pro 3rd gen, It somehow doesn't happen to him on both same network and on his wifi.
> 6. A windows device with a core i7 5500u and 32gbs of ram. The chunks do load on lan but really slowly (probably my device lol) and on another network it sometimes sends me into the void or chunks don't load at all.
> 7. An iPhone 12 the regular version, Chunks will only load on another network for some reason and stop after loading a few chunks on my own network that the server is connected to
> 8. 2 more android devices (one with a Snapdragon 665 and 4gbs of ram and a exynoss 850 with 4gbs of ram) the results for these 2 were the same as device Number 2 I tested.
> 9. Xbox series X and PS4. The chunks load correctly and smoothly on this and another network with the only issue being it has a bit of lag spikes (that might just be me on 2.4g wifi though)
> 
> 10 Nintendo switch. Performance is exceptional (probably just the switch not being well optimized) chunks wont load when connected on another network or it drops me in an empty unloaded place however on Lan it will load just fine.
> 
> 11. An old hp 15 notebook with a Core i3 5600u and Intel HD graphics 16gbs of ram. Chunks load fine on lan and surprisingly on wireless 2g it does better. When connected to another network it doesn't load some chunks and skips some chunks too.

Did you reduce the render distance on all these devices to the minimum (5 chunks)?

**Comment by YaBoiEddyC** [comment]
Oh I forgot to say, Only device 1, 3 and 5 didn't have the chunk loading issue every other one did have some sort of issue on either the same network as the server or connected to another network

My server is quite modded with Fabric MC and this was tested on both modded and unmodded
Mods:
Fabric API
Lithium
Starlight
C2ME
VMP
Krypton
Ferritecore
All of yungs better mods
Floodgate and geyser
MemoryLeakFix
Collective+player heads
Stardust Labs mods (Terralith, Incendium, nullscape, structory, towers)
Repurposed Structures
ChoiceTheorem's Overhauled Village
Better End cities Datapacks 
Forever babies Datapack.
Flan Claim Mod 
Dynmap Fabric
Discord4Fabric

Server Specs: Ryzen 7 5800h 3.2ghz base and 4.4 max, and 64gbs of ram(18gbs dedicated to the server), 1tb NVME 4 SSD,
Lan: 1 gigabit down and 35up.

**Comment by YaBoiEddyC** [comment]
@Kas-tle I tested each one to the minimum and max of the server render distance

**Comment by Kas-tle** [comment]
> @Kas-tle I tested each one to the minimum and max of the server render distance

So you didn't set the render distance of the client to 5 chunks?

**Comment by YaBoiEddyC** [comment]
@Kas-tle Yes I did on the client

**Comment by iDarknessSam** [comment]
mine is having the same problem too on Aternos using the Spigot to run the plugin, my friend on Bedrock can't play with the world don't loading all the time. i'm on the 1.20 version of the Minecraft

https://dump.geysermc.org/LNz5PK6mnuq3HxLTQR5K3GLxe87eBd4B

**Comment by iDarknessSam** [comment]
> mine is having the same problem too on Aternos using the Spigot to run the plugin, my friend on Bedrock can't play with the world don't loading all the time. i'm on the 1.20 version of the Minecraft
> 
> https://dump.geysermc.org/LNz5PK6mnuq3HxLTQR5K3GLxe87eBd4B

Actually...I tried with others devices and all of them are having the same problem..no matter what render I put on it the world stop to render after certain point. some of my friends on bedrock are being sad with this

**Comment by iDarknessSam** [comment]
> 

even updating the plugin to the latest version, still nothing

**Comment by YaBoiEddyC** [comment]
@iDarknessSam PaperMC is probably the only one that doesn't have the chunk loading issue due to their chunk rewrite so if you wanna have a server with chunks loading correctly for now just use paper

**Comment by iDarknessSam** [comment]
> @iDarknessSam PaperMC is probably the only one that doesn't have the chunk loading issue due to their chunk rewrite so if you wanna have a server with chunks loading correctly for now just use paper

but the plugins from Spigot works on PaperMC without any issues?

**Comment by iDarknessSam** [comment]
> @iDarknessSam PaperMC is probably the only one that doesn't have the chunk loading issue due to their chunk rewrite so if you wanna have a server with chunks loading correctly for now just use paper

![photo_2023-09-10_21-24-57](https://github.com/GeyserMC/Geyser/assets/59544278/d6903452-7591-4a94-a962-79b047e84f60)
![Screenshot_20230910_212545](https://github.com/GeyserMC/Geyser/assets/59544278/7c92dcab-4928-4409-bf44-1144d74005f4)

I installed PaperMC here but the problem still persist in all devices..

**Comment by YaBoiEddyC** [comment]
@iDarknessSam Which version are you using

**Comment by iDarknessSam** [comment]
> @iDarknessSam Which version are you using

1.20.1 version of Java and the latest version of GeyserMC on PaperMC

**Comment by YaBoiEddyC** [comment]
@iDarknessSam yeah I tested it again on most of the devices it seems like it's affecting paper too now

**Comment by iDarknessSam** [comment]
> > @iDarknessSam PaperMC is probably the only one that doesn't have the chunk loading issue due to their chunk rewrite so if you wanna have a server with chunks loading correctly for now just use paper
> 
> ![photo_2023-09-10_21-24-57](https://user-images.githubusercontent.com/59544278/266881472-d6903452-7591-4a94-a962-79b047e84f60.jpg) ![Screenshot_20230910_212545](https://user-images.githubusercontent.com/59544278/266881474-7c92dcab-4928-4409-bf44-1144d74005f4.png)
> 
> I installed PaperMC here but the problem still persist in all devices.. https://dump.geysermc.org/6HCxNvCfKHc7ReyOtTuVtx25cWe3AGHh

https://github.com/GeyserMC/Geyser/assets/59544278/5cc1eb6b-9774-4404-95f5-9d3f379d02bd

**Comment by YaBoiEddyC** [comment]
@iDarknessSam Maybe its a client issue since some versions of Minecraft on android are like ARM7, ARM8, ARM64 etc. because on my newer phone that's ARM9 doesn't have the issue but my ARM7 tablet does

**Comment by iDarknessSam** [comment]
> @iDarknessSam Maybe its a client issue since some versions of Minecraft on android are like ARM7, ARM8, ARM64 etc. because on my newer phone that's ARM9 doesn't have the issue but my ARM7 tablet does

well...I guess so, since it only happens on Bedrock and not Java
I even tested with waydroid that use the x86_64 (not sure if it really x86_64) and the same problem happens on it too...
but onee thing is clear, maybe this problem is becoming more common since a lot of people are having this stuff too.

**Comment by YaBoiEddyC** [comment]
@iDarknessSam Yeah but for me on Windows it does just fine unless its my older laptops and it only happens on mobile for me if its an ARMv7 device or lower.

**Comment by iDarknessSam** [comment]
> @iDarknessSam Yeah but for me on Windows it does just fine unless its my older laptops and it only happens on mobile for me if its an ARMv7 device or lower.

I hope this problem get fixed any time soon..

**Comment by iDarknessSam** [comment]
I updated to the latest version of Minecraft java and the Geyser too, and the problem is still happening

https://dump.geysermc.org/GOTGWfgSjfqNYj8dujdXS7QLbyX4Hjfw
![photo_2023-10-20_22-39-36](https://github.com/GeyserMC/Geyser/assets/59544278/a57fbf99-d952-48b6-9fc9-9ca64b2a187b)
![photo_2023-10-20_22-39-37](https://github.com/GeyserMC/Geyser/assets/59544278/de384727-5eb5-4952-8406-719b3a9f4ee6)
![photo_2023-10-20_22-39-38](https://github.com/GeyserMC/Geyser/assets/59544278/7dc42406-2822-43d6-aed0-d2bab00fa6db)

**Comment by iDarknessSam** [comment]
maybe the config folder can influence in some way?

**Comment by YaBoiEddyC** [comment]
I was reading through the wiki just recently and it might be due to bedrock being 32 bit or something like that can't confirm tho
[Screenshot_20231103-083051.png](https://github.com/GeyserMC/Geyser/assets/93556863/44f31ed1-b0a4-4d7e-9184-296dc396cc09)

**Comment by iDarknessSam** [comment]
> I was reading through the wiki just recently and it might be due to bedrock being 32 bit or something like that can't confirm tho [Screenshot_20231103-083051.png](https://github.com/GeyserMC/Geyser/assets/93556863/44f31ed1-b0a4-4d7e-9184-296dc396cc09)

I don't think that is causing the issue....maybe the plugin for itself can solve this problem or something like that. However, one thing that I was thinking that maybe are causing this render problem is probably the world size, I mean, I noticed that people who is not having this render problem the world of the server is pretty small with not so much chunks, but when the world is more than 1 GB the problem start to appear causing some problems at the render, so maybe the issue is because of the world size?

**Comment by iDarknessSam** [comment]
someone found any solution?

**Comment by Kas-tle** [comment]
@iDarknessSam you are likely experiencing this bug given your CPU and Java version: https://github.com/GeyserMC/Geyser/issues/2861
Please update Java on your server running Geyser to the latest (should be 17.0.9).

**Comment by Kas-tle** [comment]
If anyone is still experiencing this issue, please post a current Geyser dump.

**Comment by StKillReal** [comment]
![image](https://github.com/GeyserMC/Geyser/assets/60330193/d6d14b1f-d93c-49c6-b32f-41ae294bc3cc)
https://dump.geysermc.org/rEJnhaO34hy4uWnI3MAXHstlVJonG0sD

Still have the issue, but it looks like periodical one and I really can't understad what affect on it. For example: now bedrock players can fly around and chunk loading is fine, but 30 mins ago they can teleport somewhere and load about 300 blocks from the spot, then chunks stops to load until player teleports to the other spot of the map, where he can load 200-300 blocks and loading stops again. There is no any errors in console of the server/velocity

**Comment by onebeastchris** [comment]
Hello! Can you try the builds here and see if it fixes your issue? https://github.com/GeyserMC/Geyser/actions/runs/9607076730

**Comment by Hana2736** [comment]
> Hello! Can you try the builds here and see if it fixes your issue? https://github.com/GeyserMC/Geyser/actions/runs/9607076730

No luck for me on Geyser-Velocity and PaperMC 1.21

**Comment by onebeastchris** [comment]
Hey, could those affected try the builds found here: https://github.com/GeyserMC/Geyser/actions/runs/9607076730? These should remedy the issue. 

@Hana2736 for you specifically, if this build also does not help, please run `geyser dump` in the server console, and send the link you get here.

**Comment by Hana2736** [comment]
This build is for 1.21.0/1 and I'm not able to downgrade to that version through the bedrock launcher

**Comment by onebeastchris** [comment]
Oops, linked the wrong build – try https://github.com/GeyserMC/Geyser/actions/runs/9922563154

**Comment by Hana2736** [comment]
![Well, not quite](https://github.com/user-attachments/assets/d6923c99-8a63-43f9-beed-8131256b2326)

[/geyser dump](https://dump.geysermc.org/ZY56MjSG0nPPC4N35nwPgMGWy6XlnrZJ)

**Comment by Hana2736** [comment]
The game seems to act like the blocks are there: i can still dig, build, and see entities. A chunk or so beyond this blocks me with an invisible wall

**Comment by Camotoy** [comment]
And what is the server render distance and client render distances set to?

**Comment by Hana2736** [comment]
Client is set to 10 chunks, server is set to 64.

**Comment by yqs112358** [comment]
Encounted same problem.
Version: Minecraft 1.21 Fabric + Geyser-Fabric 2.4.0 build 606 & Bedrock Android 1.21
Nothing was output in server console when this bug occurs. Set client view-distance to the same as server's cannot fix the bug completely.

**Comment by seanosh** [comment]
Geyser dump [here](https://dump.geysermc.org/MIEuSwZqQ0ioblnxIu2HyPG3an8Jje6H), server render distance is 10 (doesn't look like this is included in the dump, unless I'm missing it, but can see it's being sought after in a few comments).

**Client**
Minecraft: Bedrock iOS 1.21.2
Render distances attempted: 11, 10
Result: **Seemingly works!**

"Seemingly" because I'm reading the comments/suggestions here and readily encountered the issue using `Build #609 7/16/2024` less than 12 hours ago.

After applying the `Geyser-Fabric` artifact linked [here](https://github.com/GeyserMC/Geyser/issues/3809#issuecomment-2227544733), I can no longer just as readily reproduce the issue. If there's a more "scientific" method to test that would be helpful, lemme know.

**Comment by Hana2736** [comment]
It happens sporadically when exploring on foot, but I have a homes plugin on the server and teleporting ~500 blocks away seems to reliably break it.

**Comment by wryel2000** [comment]
I have this problem too, not have solution ? bruuuuh

**Comment by mrchilliballs** [comment]
I had been having this issue in my server for a while, with mobile players mainly complaining. The build seems to work perfectly, with no chunk loading issues when boating or using a minecart.

**Comment by Stuart-Johnson** [comment]
Just adding my two cents here. I had this problem, and confirmed that the issue seemed to be only happening if the client render distance was less than the server render distance.

Since I didn't want to lower my server render distance, I use this plugin now: https://modrinth.com/plugin/seemore

Works like a charm as far as I can tell, but I'd love to see if it resolves the issue for anyone else.

**Comment by wryel2000** [comment]
> Apenas adicionando meus dois centavos aqui. Eu tive esse problema e confirmei que o problema parecia estar acontecendo apenas se a distância de renderização do cliente fosse menor que a distância de renderização do servidor.
> 
> Como não queria diminuir a distância de renderização do meu servidor, agora uso este plugin: https://modrinth.com/plugin/seemore
> 
> Pelo que sei, funciona perfeitamente, mas adoraria ver se isso resolve o problema para mais alguém.

u have other option to old servers ? im running um 1.16.5

**Comment by UnbeatenPlayzYT** [comment]
> Just adding my two cents here. I had this problem, and confirmed that the issue seemed to be only happening if the client render distance was less than the server render distance.
> 
> Since I didn't want to lower my server render distance, I use this plugin now: https://modrinth.com/plugin/seemore
> 
> Works like a charm as far as I can tell, but I'd love to see if it resolves the issue for anyone else.

I will try this on my Elytra server sometime and see what happens.
Thank you bro.

**Comment by Stuart-Johnson** [comment]
@UnbeatenPlayzYT I’m running paper, so yes. I haven’t updated to the latest version of paper yet, but as of a few months ago it worked fine.

As for chunky, I don’t think it matters? Chunky just does chunk creation, I don’t think it has any connection to render distance, at least in a direct technical sense.

**Comment by Jsaunders411** [comment]
This problem still happens and setting the client to match server render distance no longer helps.its getting more and more frequently and wont resolve unless  the bedrock client resets hopefully soon a resolution can be found.

**Comment by UnbeatenPlayzYT** [comment]
> This problem still happens and setting the client to match server render distance no longer helps.its getting more and more frequently and wont resolve unless the bedrock client resets hopefully soon a resolution can be found.

Have you tried the 'See More' plugin mentioned above?

**Comment by Jsaunders411** [comment]
> > This problem still happens and setting the client to match server render distance no longer helps.its getting more and more frequently and wont resolve unless the bedrock client resets hopefully soon a resolution can be found.
> 
> Have you tried the 'See More' plugin mentioned above?

No i havent does it work seems like any time they use a boat elytra or minecart any fast movement. Ised to be you could have them set the render distance to match it problem would ease up now it seems alot worse.

**Comment by CherubicBoot** [comment]
any update on this? and is there an alternative for seemore for fabric server?
